### PR TITLE
Damgard-Fujisaki associated ZKPs

### PR DESCRIPTION
**Pull request by miha-stopar**
DF scheme fixed. Two associated ZKPs implemented:
- Proof that you know how to open a commitment: given a commitment c, prove that you know x, r such that c = g^x * h^r (mod N).
- Proof that x3 = x1*x2 given three commitments c1, c2, c3 where:
  c1 = g^x1 * h^r1 (mod N)
  c2 = g^x2 * h^r2 (mod N)
  c3 = g^x3 * h^r3 (mod N)

### CODE REVIEW

File: crypto/zkp/primitives/commitments/damgard-fujisaki_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/crypto/commitments"
+	"github.com/xlab-si/emmy/crypto/common"
+	"math/big"
+	"testing"
+)
+
+// TestProveDFCommitmentOpening demonstrates how to prove that you can open DamgardFujisaki commitment.
+func TestProveDFCommitmentOpening(t *testing.T) {
```
Comment (mancabizjak) [inline crypto/zkp/primitives/commitments/damgard-fujisaki_test.go]: I suggest renaming this test to something that includes `DamgardFujisaki`. For instance, `TestOpenDamgardFujisakiCommitment` (or whatever you find most appropriate). It produces a clearer output when the tests are actually run.

File: crypto/zkp/primitives/commitments/damgard-fujisaki_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	challenge := verifier.GetChallenge()
+	s1, s2 := prover.GetProofData(challenge)
+	proved := verifier.Verify(s1, s2)
+
+	assert.Equal(t, true, proved, "DamgardFujisaki opening proof failed.")
+}
+
+// TestProveDFCommitmentMultiplication demonstrates how to prove that for given commitments
+// c1 = g^x1 * h^r1, c2 = g^x2 * h^r2, c3 = g^x3 * h^r3, it holds x3 = x1 * x2
+func TestProveDFCommitmentMultiplication(t *testing.T) {
```
Comment (mancabizjak) [inline crypto/zkp/primitives/commitments/damgard-fujisaki_test.go]: I suggest renaming this test to something that includes `DamgardFujisaki`. For instance, `TestProveDamgardFujisakiCommitmentMultiplication`. It is indeed longer but produces a clearer output when the tests are actually run.

**Comment by miha-stopar** [comment]
I think I addressed all comments. I haven't added DamgardFujisaki New* function because it is only to be used internally when initializing DamgardFujisakiCommitter and DamgardFujisakiReceiver, instead I made it internal (now damgardFujisaki).

**Comment by mancabizjak** [comment]
Thanks, great. 

I have two more comments (sorry, I didn't notice the first one before)
* please create test files `damgard-fujisaki_multiplication_test.go` and `damgard-fujisaki_opening.go` and put each unit test to appropriate file. And remove the file `damgard-fujisaki_test.go` that contains both of the tests. 
* Please also update comments above tese two tests, so that correct names are used (the old ones are still present).

**Pull request merged**